### DISCUSSION TITLE
possible to target brew main repo?

### DISCUSSION THREAD

**Discussion by maxandersen** [jreleaser/jreleaser#878] [General]
currently jreleaser does brew as "taps" which is nice and independent but i'm curious if could make use of jreleaser to make pull requests to https://github.com/Homebrew/homebrew-core/tree/master/Formula instead of commit/push to a tap repo ?

I kinda feel it might already be possible but haven't succeeded in getting somethig working yet.

**Comment by aalmiray** [comment]
We'll have to do some research on the reqs & constraints to build and push core Formulae.

**Reply by aalmiray** [reply]
Additional information:

https://docs.brew.sh/Acceptable-Formulae
https://docs.brew.sh/Acceptable-Casks

Distributions of type `BINARY`, `NATIVE_IMAGE` should go into [homebrew/cask](https://github.com/Homebrew/homebrew-cask).

**Comment by maxandersen** [comment]
now that jbang got contributed by community to homebrew-core I now really would like to find a way to do this.

afaics. if I somehow could just take the formula generated for homebrew-tap and put into homebrew-core repo and submit pull request like 'article' can do it all would be good.

Any way to do that ?

maybe allow to have 2 `brew` entries one targeting core homebrew repo the other the default homebrew-tap?

**Reply by aalmiray** [reply]
Possible scenarios:
1. formula for local tap only.
2. shared formula between local tap and core. 
3. formula for local tap. different formula for core.
4. core formula only. 

At the moment JReleaser only supports the first case. AFAICT updating a core formula can be done with a couple of existing GH actions. However, the formula mist match the requirements hence generating it from a template make sense. 

If JBang’s brew template is updated to match use case 2 then it would a matter of merging the tap into a fork of the core formulae and then send a PR. The commit step can be performed by JReleaser, the update and PR by other GH actions. 

OTOH if the brew template is different than what’s expected from core and you still want 2 different formulae (use case 3) then we may need to add code to support it. This btw would be my preferred option for JReleaser’s formula as the tap installs `jreleaser-standalone` binaries (jlink) while the core formula installs the universal binary and relies on a dependency to the `openjdk` package. Also, the tap has all history while the core formula only has 1 or 2 versions so far. 

Use case 4 can also be realized today as it’s a matter of changing the template to match core expectations. The last step is sending a PR to the core repo. 

The trick now would be to support more than 1 formula template for a given distribution. with each template having shared and unique settings, for example each formula goes to a different repository.

**Reply by aalmiray** [reply]
More info about the additional GH actions that can update an existing core formula https://github.com/jreleaser/jreleaser.github.io/pull/42#issuecomment-1368134466
